drivers/clock_management/CMakeLists.txt

```diff
@@ -0,0 +1,16 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_include_directories(.)
+zephyr_library()
+
+zephyr_library_sources(clock_management_common.c)
+zephyr_library_sources_ifdef(CONFIG_CLOCK_MANAGEMENT_FIXED_SOURCE fixed_clock_source.c)
+zephyr_library_sources_ifdef(CONFIG_CLOCK_MANAGEMENT_SOURCE clock_source.c)
+# Header for common clock drivers
+add_clock_management_header(clock_management_drivers.h)
+
+add_subdirectory(nxp_syscon)
+
+# Include headers for all clock management drivers, registered with add_clock_management_header
+zephyr_library_compile_options("-include$<JOIN:$<TARGET_PROPERTY:clock_management_header_target,COMPILE_DEFINITIONS>,;-include>")
```

Comment:
When I have this in application level CMakeLists.txt.
```
execute_process(COMMAND hg identify
    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
    OUTPUT_VARIABLE BUILD_HASH)
string(STRIP ${BUILD_HASH} BUILD_HASH)

add_definitions(-DBUILD_HASH="${BUILD_HASH}")
```

I get an error for this

```
gcc-arm-none-eabi/12.2-Rel1/arm-gcc-linux/bin/arm-none-eabi-gcc -DKERNEL -DK_HEAP_MEM_POOL_SIZE=4096 -DPB_MAX_REQUIRED_FIELDS=64 -D_ANSI_SOURCE -D__LINUX_ERRNO_EXTENSIONS__ -D__PROGRAM_START -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I/redacted/everett_ctrl_app/build/zephyr/include/generated/zephyr -Izephyr/include -I/redacted/everett_ctrl_app/build/zephyr/include/generated -I/redacted/soc/meta/everett -Izephyr/lib/libc/newlib/include -Izephyr/lib/posix/options/getopt -Izephyr/drivers -I/redacted/.. -I/redacted/soc/meta/everett/m55/. -Izephyr/subsys/rtio -Izephyr/drivers/clock_management/. -Imodules/hal/cmsis/CMSIS/Core/Include -Izephyr/modules/cmsis/. -I/redacted/everett_ctrl_app/build/modules/libmetal/libmetal/lib/include -Izephyr/../modules/lib/nanopb -Imodules/lib/open-amp/open-amp/lib/include -isystem zephyr/lib/libc/common/include -Wall -Wextra -Wno-unused-parameter -Wno-cast-function-type -Wno-sign-compare -Werror=double-promotion -Wdouble-promotion -fdebug-prefix-map=/redacted/everett_ctrl_app/build=. -fdebug-prefix-map=/redacted/everett_ctrl_app/=/ -ffile-prefix-map=/=/ -fdebug-prefix-map=/=/ -Wshadow -fno-strict-aliasing -Os -imacros /redacted/everett_ctrl_app/build/zephyr/include/generated/zephyr/autoconf.h -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m55+nodsp -mthumb -mabi=aapcs -mfp16-format=ieee --sysroot=arm-gcc-linux/arm-none-eabi -imacros zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wformat -Wformat-security -Wno-format-zero-length -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/redacted/everett_ctrl_app=CMAKE_SOURCE_DIR -fmacro-prefix-map=zephyr=ZEPHYR_BASE -ffunction-sections -fdata-sections -specs=nano.specs -D_POSIX_THREADS -std=c11 -includezephyr/drivers/clock_management/clock_management_drivers.h -includeBUILD_HASH=\"1b44ca44ae32+\" -MD -MT zephyr/drivers/clock_management/CMakeFiles/drivers__clock_management.dir/clock_management_common.c.obj -MF zephyr/drivers/clock_management/CMakeFiles/drivers__clock_management.dir/clock_management_common.c.obj.d -o zephyr/drivers/clock_management/CMakeFiles/drivers__clock_management.dir/clock_management_common.c.obj -c zephyr/drivers/clock_management/clock_management_common.c
<command-line>: fatal error: BUILD_HASH="1b44ca44ae32+": No such file or directory
```

Reply:
Good catch. It looks like this is caused because I was using the `COMPILE_DEFINITIONS` property for the target to hold the include headers- instead of overloading that property (which can be modified globally with `add_definitions`, as you found) we can use the `SOURCES` property for this custom target to store that list. Made that change in the latest push, and it seems to fix the issue.